Content.Server/_Goobstation/Devil/Contract/DevilContractSystem.cs

```diff
@@ -0,0 +1,506 @@
+using Content.Shared.Mindshield.Components;
+using Content.Shared.Nutrition;
+using Content.Shared.Paper;
+using Content.Shared.Popups;
+using Content.Shared.Silicons.Borgs.Components;
+using Content.Shared.Verbs;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
+using Content.Shared._EE.Silicon.Components;
+
+namespace Content.Server._Goobstation.Devil.Contract;
+
+public sealed partial class DevilContractSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPopupSystem _popupSystem = null!;
+    [Dependency] private readonly DamageableSystem _damageable = null!;
+    [Dependency] private readonly HandsSystem _hands = null!;
+    [Dependency] private readonly SharedAudioSystem _audio = null!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = null!;
+    [Dependency] private readonly BodySystem _bodySystem = null!;
+    [Dependency] private readonly IRobustRandom _random = null!;
+    [Dependency] private readonly SubdermalImplantSystem _implant = null!;
+    [Dependency] private readonly PolymorphSystem _polymorph = null!;
+    [Dependency] private readonly ExplosionSystem _explosion = null!;
+    [Dependency] private readonly MindSystem _mind = null!;
+
+    private ISawmill _sawmill = null!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        InitializeRegex();
+        InitializeSpecialActions();
+
+        SubscribeLocalEvent<DevilContractComponent, BeingSignedAttemptEvent>(OnContractSignAttempt);
+        SubscribeLocalEvent<DevilContractComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<DevilContractComponent, GetVerbsEvent<AlternativeVerb>>(OnGetVerbs);
+        SubscribeLocalEvent<DevilContractComponent, SignSuccessfulEvent>(OnSignStep);
+        SubscribeLocalEvent<DevilContractComponent, AfterFullyEatenEvent>(OnEaten);
+
+        _sawmill = Logger.GetSawmill("devil-contract");
+    }
+
+    private readonly Dictionary<LocId, Func<DevilContractComponent, EntityUid?>> _targetResolvers = new()
+    {
+        // The contractee is who is making the deal.
+        ["devil-contract-contractee"] = comp => comp.Signer,
+        // The contractor is the entity offering the deal.
+        ["devil-contract-contractor"] = comp => comp.ContractOwner,
+    };
+
+    private Regex _clauseRegex = null!;
+
+    private void InitializeRegex()
+    {
+        var escapedPatterns = _targetResolvers.Keys.Select(locId => Loc.GetString(locId)).ToList(); // malicious linq and regex
+        var targetPattern = string.Join("|", escapedPatterns);
+
+        _clauseRegex = new Regex($@"^\s*(?<target>{targetPattern})\s*:\s*(?<clause>.+?)\s*$",
+            RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.Multiline);
+    }
+    private void OnGetVerbs(Entity<DevilContractComponent> contract, ref GetVerbsEvent<AlternativeVerb> args)
```

Comment:
Blank line between methods
```suggestion
    }

    private void OnGetVerbs(Entity<DevilContractComponent> contract, ref GetVerbsEvent<AlternativeVerb> args)
```